File: website/docs/concepts/typegraph/triggers.py
# skip:start
from typegraph import TypeGraph, policies, t
from typegraph.runtimes.http import HTTPRuntime

# skip:end
with TypeGraph("triggers") as g:
    # skip:start
    public = policies.public()
    http = HTTPRuntime("https://random.org/api")
    # skip:end
    # ...
    g.expose(
        flip=http.get("/flip_coin", t.struct({}), t.enum(["head", "tail"])),
        default_policy=[public],
    )

File: website/typegraphs/triggers.ts
// skip:start
import { Policy, t, typegraph } from "@typegraph/sdk/index.js";
import { HttpRuntime } from "@typegraph/sdk/runtimes/http.js";

// skip:end

typegraph({
  name: "triggers",
}, (g) => {
  // skip:start
  const pub = Policy.public();
  const http = new HttpRuntime("https://random.org/api");
  // skip:end
  // ...
  g.expose({
    flip: http.get(t.struct({}), t.enum_(["head", "tail"]), {
      path: "/flip_coin",
    }),
  }, pub);
});
